src/FloorSubsystem/FloorSubsystem.java

```diff
@@ -0,0 +1,94 @@
+package FloorSubsystem;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import common.SimulationFloorInputData;
+import common.requests.JobRequest;
+import common.requests.MessageChannel;
+
+
+/**
+ * This class simulates the FloorSubsystem thread
+ * 
+ * @author Favour
+ * @author Delight
+ */
+public class FloorSubsystem implements Runnable{
+	
+	private String inputFileName;
+	private ArrayList<SimulationFloorInputData> floorDataCollection;
+	private FloorInfo floor;
+	private MessageChannel floorMessageChannel;
+
+	/**
+	 * This is the default constructor of the class
+	 * 
+	 * @param inputFileName - The input text file
+	 */
+	public FloorSubsystem(String inputFileName) {
+		this.inputFileName = inputFileName;
+		floorDataCollection = new ArrayList<>();
+		floor = new FloorInfo();
+		floorMessageChannel = new MessageChannel();
+	}
+	
+	/**
+	 * This method reads in the input text file and converts
+	 * it to SimulationFloorInputData objects as needed
+	 * 
+	 */
+	private void readInputFile() {
+		BufferedReader bufferedReader = null;
+		
+		try {
+			bufferedReader = new BufferedReader(new FileReader(inputFileName));
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
```

Comment:
Would we want to throw an exception? 

I do not really see the point of an elevator progressing if we cannot read the file.